gems/pending/util/mount/miq_generic_mount_session.rb

```diff
@@ -451,14 +451,18 @@ def with_mounted_exception_handling
     raise
   end
 
-  def self.raw_disconnect(mnt_point)
-    case Sys::Platform::IMPL
-    when :macosx
-      runcmd("sudo umount #{mnt_point}")
-    when :linux
-      runcmd("umount #{mnt_point}")
-    else
-      raise "platform not supported"
+  class << self
+    protected
```

Comment:
I wasn't quite sure how to tackle this. The class methods were originally beneath [`protected`](https://github.com/ManageIQ/manageiq/pull/12041/files/980057de1f7f59e96d2d9fcc9f392238f8ee43b5#diff-8b37bf510f735b78222889d7d5763cd2R427), so I created the singleton class and added `protected` inside it. Happy to move these to `private_class_method`s though if that makes sense.